wireless/ieee802154/mac802154_internal.h

```diff
@@ -510,62 +511,12 @@ void mac802154_notify(FAR struct ieee802154_privmac_s *priv,
 
 /* General helpers **********************************************************/
 
-#define mac802154_givesem(s) nxsem_post(s)
-
-static inline int mac802154_takesem(sem_t *sem, bool allowinterrupt)
-{
-  if (allowinterrupt)
-    {
-      return nxsem_wait(sem);
-    }
-  else
-    {
-      return nxsem_wait_uninterruptible(sem);
-    }
-}
-
-#ifdef CONFIG_MAC802154_LOCK_VERBOSE
-#define mac802154_unlock(dev) \
-  mac802154_givesem(&dev->exclsem); \
-  wlinfo("MAC unlocked\n");
-#else
-#define mac802154_unlock(dev) \
-  mac802154_givesem(&dev->exclsem);
-#endif
-
-#define mac802154_lock(dev, allowinterrupt) \
-  mac802154_lockpriv(dev, allowinterrupt, __FUNCTION__)
-
-static inline int
-mac802154_lockpriv(FAR struct ieee802154_privmac_s *dev,
-                   bool allowinterrupt, FAR const char *funcname)
-{
```

Comment:
I think this changes the logic significantly since driver was allowing to handle signal interrupt. I'm not sure if such situation is relevant for kernel at all and this driver does not seems to run as a user space application. So anyway seems to be fine to use nxmutex instead

Reply:
Yes, I think it's very dangerous to allow user cancel the kernel work and then remove the possibility.